lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -538,3 +543,182 @@ def test_move_users_to_same_cohort(self):
             ],
             verify_order=False
         )
+
+
+@patch('instructor_task.tasks_helper.DefaultStorage', new=MockDefaultStorage)
+class TestGradeReportCertificateInfo(TestReportMixin, InstructorTaskModuleTestCase):
+    """
+    Test that grade report has correct user certificate information.
+    """
+    def setUp(self):
+        super(TestGradeReportCertificateInfo, self).setUp()
+
+        self.initialize_course()
+
+        self.create_problem()
+
+        self.csv_cert_info_header = [
+            'Enrollment Track',
+            'Verification Status',
+            'Certificate Eligible',
+            'Certificate Delivered',
+            'Certificate Type'
+        ]
+
+    def create_problem(self, problem_display_name='test_problem', parent=None):
+        """
+        Create a multiple choice response problem.
+        """
+        if parent is None:
+            parent = self.problem_section
+
+        factory = MultipleChoiceResponseXMLFactory()
+        args = {'choices': [False, True, False]}
+        problem_xml = factory.build_xml(**args)
+        ItemFactory.create(
+            parent_location=parent.location,
+            parent=parent,
+            category="problem",
+            display_name=problem_display_name,
+            data=problem_xml
+        )
+
+    def generate_certificate(self, user, course_id, status, mode):
+        """
+        Generate certificates for users.
+        """
+        GeneratedCertificateFactory.create(
+            user=user,
+            course_id=course_id,
+            status=status,
+            mode=mode
+        )
+
+    def user_in_whitelist(self, user, course_id, state):
+        """
+        Set a users whitelist state.
+        """
+        cert_whitelist, _ = CertificateWhitelist.objects.get_or_create(user=user, course_id=course_id)
+        cert_whitelist.whitelist = state
+        cert_whitelist.save()
+
+    def user_is_embargoed(self, user, is_embargoed):
+        """
+        Set a users emabargo state.
+        """
+        user_profile = UserFactory(username=user.username, email=user.email).profile
+        user_profile.allow_certificate = not is_embargoed
+        user_profile.save()
+
+    def set_user_verification_status(self, user, status):
+        """
+        Set a users verification status.
+        """
+        attempt, _ = SoftwareSecurePhotoVerification.objects.get_or_create(user=user)
```

Comment:
Can you also create a factory for `SoftwareSecurePhotoVerification` and use it instead?